rsmtool/configuration_parser.py

```diff
@@ -44,33 +47,112 @@
                                    default_feature_sign)
 
 
+
+def deprecated_positional_argument():
+    """ This decorator allows for
+    backwards compatibility of the Configuration class
+    where filepath could be passed as a positional argument
+    and no configdir was set
+    Based on https://stackoverflow.com/a/49802489"""
+    def deco(f):
+        @functools.wraps(f)
+        def wrapper(*args, **kwargs):
+            # if we received two positional arguments
+            if len(args) > 2:
+                # if we also received a keyword argument for filepath
+                # or configdir, raise an error
+                if 'filename' in kwargs:
+                    raise ValueError("Configuration received both "
+                                     "positional argument for filepath "
+                                     "and keyword argument "
+                                     "for filename.")
+                if 'configdir' in kwargs:
+                    raise ValueError("Configuration received both "
+                                     "positional argument for filepath "
+                                     "and keyword argument for configdir")
+                # raise deprecation warning
+                warnings.warn("Starting from RSMTool 8.0, you will need to "
+                              "specify configdir as a keyword-only argument "
+                              "when initializing a Configuration object ",
+                              DeprecationWarning)
```

Comment:
It will raise a warning, if you pass a second positional argument to `Configuration` (current API): e.g. `Configuration(dict, 'path_to_file'). It will raise a warning if you pass a second positional argument plus any of the new keyword arguments. In other words, if you combine old and new API.

Reply:
You can have a look at the tests too.